NumLib/ODESolver/ConvergenceCriterionResidual.h

```diff
@@ -39,6 +39,7 @@ class ConvergenceCriterionResidual final : public ConvergenceCriterion
     void checkResidual(const GlobalVector& residual) override;
 
     void preFirstIteration() override { _is_first_iteration = true; }
+    void setNoFirstIteration() override { _is_first_iteration = false; }
```

Comment:
``reset()`` sets ``_satisfied = true``, which is not proper for the current implementation.